### PR TITLE
[TIMOB-23424] Android: Support presentInviteDialog

### PR DESCRIPTION
**Pull request by hansemannn**
JIRA: https://jira.appcelerator.org/browse/TIMOB-23424

### CODE REVIEW

File: apidoc/Facebook.yml [hunk 1]
```diff
@@ -661,7 +661,7 @@ methods:
             A dictionary object containing required and optional parameters.
         type: InviteDialogParams
     since: "5.4.0"
-    platforms: [iphone, ipad]
+    platforms: [iphone, ipad, android]
```
Comment (hieupham007) [inline apidoc/Facebook.yml:664]: Is this still going into 5.4 or 6.0?

Comment (hansemannn) [inline apidoc/Facebook.yml:664]: Feature-freeze is tomorrow, so if you can approve it until tomorrow, we could deliver the new API cross-platform from the beginning. But it needs a FT

**Comment by cheekiatng** [comment]
@hieupham007 I can make an exception for this one for parity sake since ios component has already been merged and implemented. So if you can review soonest it'll be great.

**Comment by hieupham007** [comment]
Getting this error running the demo app: 
[ERROR] :  TiBaseActivity: (main) [1,97] Error dispatching lifecycle event: Both context and applicationId must be non-null
[ERROR] :  TiBaseActivity: java.lang.IllegalArgumentException: Both context and applicationId must be non-null
[ERROR] :  TiBaseActivity:  at com.facebook.appevents.AppEventsLogger.deactivateApp(AppEventsLogger.java:360)
[ERROR] :  TiBaseActivity:  at com.facebook.appevents.AppEventsLogger.deactivateApp(AppEventsLogger.java:338)
[ERROR] :  TiBaseActivity:  at facebook.TiFacebookModule.onPause(TiFacebookModule.java:143)
[ERROR] :  TiBaseActivity:  at org.appcelerator.titanium.TiLifecycle.fireLifecycleEvent(TiLifecycle.java:151)
[ERROR] :  TiBaseActivity:  at org.appcelerator.titanium.TiBaseActivity.onPause(TiBaseActivity.java:1208)
[ERROR] :  TiBaseActivity:  at org.appcelerator.titanium.TiLaunchActivity.onPause(TiLaunchActivity.java:337)
[ERROR] :  TiBaseActivity:  at android.app.Activity.performPause(Activity.java:6348)
[ERROR] :  TiBaseActivity:  at android.app.Instrumentation.callActivityOnPause(Instrumentation.java:1311)
[ERROR] :  TiBaseActivity:  at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3367)
[ERROR] :  TiBaseActivity:  at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3340)
[ERROR] :  TiBaseActivity:  at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:3315)
[ERROR] :  TiBaseActivity:  at android.app.ActivityThread.-wrap13(ActivityThread.java)
[ERROR] :  TiBaseActivity:  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1355)
[ERROR] :  TiBaseActivity:  at android.os.Handler.dispatchMessage(Handler.java:102)
[ERROR] :  TiBaseActivity:  at android.os.Looper.loop(Looper.java:148)
[ERROR] :  TiBaseActivity:  at android.app.ActivityThread.main(ActivityThread.java:5417)
[ERROR] :  TiBaseActivity:  at java.lang.reflect.Method.invoke(Native Method)
[ERROR] :  TiBaseActivity:  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
[ERROR] :  TiBaseActivity:  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)

**Comment by hansemannn** [comment]
The error is, that your sample project is miconfigured. As the error says, you need to include the app-id and attach the fbProxy property to the window on Android. Check the readme for more infos on that.

**Comment by hieupham007** [comment]
Yep, just did that, now getting this:  ALERT: (KrollRuntimeThread) [2,38242] {cancelBubble=false, source=facebook.TiFacebookModule@620a60c, error=Error inviting people, bubbles=false, type=inviteCompleted, cancelled=false, success=false}

**Comment by hansemannn** [comment]
Make sure you FB-App-ID is valid and is configured for Android. I tested it on the sim without problems today (both cancel and success).

**Comment by hieupham007** [comment]
Tested with appLink: https://fb.me/868269279951930, works as expected. Got ALERT: (KrollRuntimeThread) [2,10639] {cancelBubble=false, source=facebook.TiFacebookModule@972cd10, bubbles=false, type=inviteCompleted, cancelled=false, success=true}

**Comment by hieupham007** [comment]
Code reviewed and functionally tested. Request **accepted**

**Pull request merged**